jaxrs/src/main/java/com/facebook/airlift/jaxrs/AuthorizationFilter.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2010 Proofpoint, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.airlift.jaxrs;
+
+import com.facebook.airlift.http.server.AuthorizationResult;
+import com.facebook.airlift.http.server.Authorizer;
+import com.facebook.airlift.http.server.HttpServerConfig;
+import com.google.common.collect.ImmutableSet;
+import com.google.inject.Inject;
+
+import javax.annotation.security.RolesAllowed;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.container.ResourceInfo;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.Provider;
+
+import java.security.Principal;
+import java.util.Optional;
+import java.util.Set;
+
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+@Provider
+public class AuthorizationFilter
+        implements ContainerRequestFilter
+{
+    private final Authorizer authorizer;
+    private final HttpServerConfig httpServerConfig;
+
+    @Context
+    private ResourceInfo resourceInfo;
+
+    @Inject
+    public AuthorizationFilter(Authorizer authorizer, HttpServerConfig httpServerConfig)
+    {
+        this.authorizer = requireNonNull(authorizer, "authorizer is null");
+        this.httpServerConfig = requireNonNull(httpServerConfig, "httpServerConfig is null");
```

Comment:
Same comment here. I would recommend passing properties that are needed instead of the entire object. You can have 2 constructors. One that accepts `HttpServerConfig httpServerConfig`, and one the accepts properties itself. The one that accepts `HttpServerConfig httpServerConfig` should have the `@Inject` annotation. This is common pattern in Presto, e.g.: https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/execution/executor/TaskExecutor.java#L169

Then the constructor that accepts the settings directly can be used in the tests.

Reply:
This looks great. I'll add unit test for this filter itself using the constructor.